### DISCUSSION TITLE
Installation

### DISCUSSION THREAD

**Discussion by kunaltelangi** [sammaji/labforall#46] [Q&A] [answered]
Could you elaborate more on installing the project with all the dependencies?
I'm done with Installing NPM using the pip

`cd experiments-website
yarn install`

kindly elaborate this part more

**Comment by sammaji** [comment, marked as answer]
Yarn is a package manager just like npm but its faster and has better caching. So I recommend using yarn.

With that said, npm is totally fine. You can use npm, no issue.

But keep in mind this project is using yarn, so delete the `yarn.lock` file and then run `npm install`. Also, do not push the `package-lock.json` (if you make any commits).

Hope that helps ✌️

**Comment by kunaltelangi** [comment]
hey, I found this project very interesting but I'm new to javascript stuff and I have a hard time finding the main file also how can I run this on the local host, I would like to add some more features like adding a home page for solar sim also adding loading animation while the site is opening.

also, I would like to add a more descriptive "how to run"  readme first file, but yeah I myself am having a hard time understanding how to run :(

**Comment by sammaji** [comment]
Yeah, actually we did the whole website in react but the simulation part is done in html/css/js. Now, due to security reasons, react does not allow injecting html and we didn't have time to migrate the simulation section to react. Hence we decided to put the code inside the index.html.

In `index.html`, line number 15 is the div inside which all the react components are put. The rest of the code is for the simulation section. My plan is to transfer the simulation code into a new file. Then I can convert it into a separate website. 

Regarding the installation part, I have a similar project and the installation process is the same. [Installation Guide](https://github.com/samyabrata-maji/magnificent-seven/blob/main/INSTALLATION.md). The main file is at `/src/main.jsx`.

And hey, thanks for taking an interest. You can create an installation guide if you want. Thanks 👍

**Reply by kunaltelangi** [reply]
if it is okay can we connect on Discord, as I'm having trouble in installation for `npm i --D yarn` also it would be more free text there

**Reply by kunaltelangi** [reply]
also, I want to add more features to Space simulation could you please mention the only files I need for space simulation,
like when I local host only space simulation should open cause seeing soo many files that I'm not familiar with, also I'm new its getting overwhelming for me to understand

**Reply by sammaji** [reply]
Ok here's my discord id -> sammaji15#9858

And regarding the simulation part, all the code for that is in the index.html and inside the simulations directory. As I said I plan to separate the simulation part into a separate project. That way I can add more simulations.

**Comment by sammaji** [comment]
Also if you're getting errors then that's because you don't have firebase api keys. I will modify the code so that you can run it locally.

**Comment by sammaji** [comment]
I have done the setup.

Once you have installed all the packages, follow these steps:
- add a `.env` file and add this line: `VITE_USE_EMULATOR=true`
- open a terminal and run `npx firebase emulators:start` to start the emulators and keep it running
- in a separate terminal, use `yarn dev` to run the project
